crates/environ/src/compile/module_types.rs

```diff
@@ -435,4 +437,19 @@ impl TypeConvert for WasmparserTypeConverter<'_> {
             UnpackedIndex::RecGroup(_) => unreachable!(),
         }
     }
+
+    fn lookup_type_index(&self, index: wasmparser::UnpackedIndex) -> EngineOrModuleTypeIndex {
```

Comment:
Could the above method delegate to this new method to start off? (looks like they both start with all the same code)

Reply:
This doesn't quite work out because while both arms of the above method start with producing an `EngineOrModuleIndex`, and that happens in the same way as this method, the arms continue by reusing intermediate steps from how the `EngineOrModuleIndex` was produced in the code that follows after the index has been constructed. This makes it subtly difficult to pull apart or reuse.